pkg/sync/sync_context.go

```diff
@@ -400,6 +410,15 @@ func (sc *syncContext) Sync() {
 
 	sc.log.WithValues("tasks", tasks).V(1).Info("Wet-run")
 	runState := sc.runTasks(tasks, false)
+
+	if sc.syncWaveHook != nil && runState != failed {
+		err := sc.syncWaveHook(phase, wave, finalWave)
+		if err != nil {
+			sc.setOperationPhase(common.OperationFailed, fmt.Sprintf("SyncWaveHook failed: %v", err))
+			runState = failed
```

Comment:
I just noticed the operation message will get clobbered by the one below. Will fix it before merging.